src/main/java/org/folio/service/orders/OrderRolloverService.java

```diff
@@ -181,33 +180,29 @@ private void throwExceptionIfOneOfFailed(AsyncResult<Void> openOrdersResult, Asy
   }
 
   private Future<Void> rolloverOrdersByFundIds(List<String> chunkFundIds, LedgerFiscalYearRollover ledgerFYRollover,
-      String systemCurrency, PurchaseOrder.WorkflowStatus workflowStatus, RequestContext requestContext) {
-
-    String query = buildOpenOrClosedOrderQueryByFundIdsAndTypes(chunkFundIds, workflowStatus, ledgerFYRollover);
+                                               String systemCurrency, PurchaseOrder.WorkflowStatus workflowStatus, RequestContext requestContext) {
+    var query = buildOpenOrClosedOrderQueryByFundIdsAndTypes(chunkFundIds, workflowStatus, ledgerFYRollover);
     var totalRecordsFuture = purchaseOrderLineService.getOrderLineCollection(query, 0, 0, requestContext)
       .map(PoLineCollection::getTotalRecords);
-
     var ctx = requestContext.getContext();
-    return totalRecordsFuture.compose(totalRecords -> ctx.<List<Future<Void>>>executeBlocking(promise -> {
-        List<Future<Void>> futures = new ArrayList<>();
+    return totalRecordsFuture.compose(totalRecords ->
+      ctx.<List<Future<Void>>>executeBlocking(promise -> {
+        var futures = new ArrayList<Future<Void>>();
```

Comment:
if you think that previous code was not so redable we have util HelperUtils::executeWithSemaphores that encupsulate complexity, but we can remain as is